the-collab-lab/tcl-33-smart-shopping-list
# Mark as Purchased for 24hrs

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

 

This issue allows the user to mark an item as purchased so the app could learn how often a user buys an item. To resolve the issue, changes were made to `App.js` and `ViewList.js`.

First, in `App.js`, a new function `checkItem`, was created. When invoked, it updates the token attribute lastPurchased, which was previously set to null, to a new date (the current date). This function is then passed down as a prop to the View List component.

Next in `ViewList.js`, a `List` component is rendered and takes in `checkItem` and `checked` as props. The new `expired(doc)` function, with a parameter of doc (the database document), has a conditional that states if it's true that lastPurchased is null, to return true. Afterwards, the `expireTime` variable becomes the time 24 hrs after the lastPurchased time in the database. The function returns true or false if the expireTime is less than the "new Date()" (current time).

A `handleChange` function says that if the function `expired(doc)` is true, then proceed with the `checkItem` function which updates the lastPurchased attribute. 

Last, the `checked` prop within the checkbox input will be activated depending on whether the `expired` function returns true or false. 

To check if the criteria was satisfied, we checked the status of the checkboxes more than 24 hours after marking the box to see if the box was still checked. We were able to resolve the issue with the help of external documents and experimentation. 

## Related Issue



Closes #8

## Acceptance Criteria


:hammer: User is able to tap a checkbox or similar UI element to mark an item in the list as purchased

:hammer:  Item should be shown as checked for 24 hours after the purchase is made (i.e. we assume the user does not need to buy the item again for at least 1 day)

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
| ✓ | :bug: Bug fix              |
| ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
| ✓ | :scroll: Docs              |

## Updates

![issue 8](https://user-images.githubusercontent.com/61391413/140652013-0ddebbc1-3608-490b-9704-d1d7e6f83d3a.gif)

## Testing Steps / QA Criteria



* From your terminal, pull down this branch with 'git pull origin af-ms-mark_as_purchased'
* Check with 'git checkout af-ms-mark_as_purchased'
* Type in 'bundle install' to see the newly added dependencies locally
* Run 'npm start' to launch the app
* View code in ViewList.js, specifically the `List` and `expired` function, and App.js for the `checkItem` function
